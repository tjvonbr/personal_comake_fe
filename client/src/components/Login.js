import React, {useState} from 'react'
import { Button, Form, Grid, Header, Image, Message, Segment, Dimmer, Loader } from 'semantic-ui-react'
import axios from 'axios';
import Welcome from '../images/bermuda/sign-in-4.png'


function Login(props) {
  const [inputData, setInputData] = useState({email: "", password:""})
  const [isLoading, setIsLoading] = useState(false)
  const [errorMessage, setErrorMessage] = useState('')
  const handleInput = e => {
    setInputData({ ...inputData, [e.target.name]: e.target.value });
  };

  const loginHandler = e => {
    setIsLoading(true)
    e.preventDefault()
    const url =
        "https://co-make.herokuapp.com/auth/login";
      axios
        .post(url, inputData)
        .then( res => {

          props.setMessage('')
          setErrorMessage('')
          console.log("Success!", res)
          props.setToken(res.data.token)
          props.setLocalId(res.data.id)
          props.setZip(res.data.zipCode)
          props.history.push("/");
          setIsLoading(false)
        })

        .catch(err => {
          console.log("Login PROBLEM0", err);
          setErrorMessage("Uh Oh, Looks like that didn't work. Try Again!")
          setIsLoading(false)
        });
  }

  return (
      <>
      <Dimmer active={ isLoading ? true : false }>
        <Loader>Loading</Loader>
      </Dimmer>
      <Grid textAlign='center' style={{ height: '100vh' }} verticalAlign='middle'>

        <Grid.Column style={{ maxWidth: 450 }}>

          <Image src={Welcome} centered size='medium' />
          <Header as='h2' color='teal' textAlign='center'>

             Log in to your account
          </Header>
          <Header as="h4" color='green' textAlign='center'>{props.message}</Header>
          <Header as="h4" color='red' textAlign='center'>{errorMessage}</Header>
          <Form size='large' onSubmit={loginHandler}>
            <Segment stacked>
              <Form.Input
              fluid icon='user'
              name="email"
              value={inputData.email}
              onChange={handleInput}
              iconPosition='left'
              placeholder='E-mail address'
               />

              <Form.Input
                fluid
                icon='lock'
                iconPosition='left'
                placeholder='Password'
                type='password'
                name="password"
                value={inputData.password}
                onChange={handleInput}
              />

              <Button type="submit" color='facebook' fluid size='large'>
                Log in
              </Button>
            </Segment>
          </Form>
          <Message>
            New to us?
            <Button className="register-button"
            onClick={()=> props.history.push('/register')}
            content='Sign Up'
            positive
            size='mini' />
          </Message>
        </Grid.Column>
      </Grid>
      </>

  )
}

export default Login
