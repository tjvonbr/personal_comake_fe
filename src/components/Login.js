import React, {useState} from 'react'
import { 
  Button, 
  Form, 
  Grid, 
  Header, 
  Image, 
  Message, 
  Segment, 
  Dimmer, 
  Loader }
from 'semantic-ui-react'
import axios from 'axios';
import Welcome from '../images/bermuda/sign-in-4.png'

function Login(props) {
  const [inputData, setInputData] = useState({username: "", password: ""});
  const [isLoading, setIsLoading] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');

  // Input handler
  const handleInput = e => {
    setInputData({ ...inputData, [e.target.name]: e.target.value });
  };

  // Submission handler
  const loginHandler = e => {
    setIsLoading(true);
    e.preventDefault();
      axios
        .post("http://localhost:4000/auth/login", inputData)
        .then(response => {
          const { id, token, zipcode } = response.data;
          window.localStorage.setItem('id', id);
          window.localStorage.setItem('token', token);
          window.localStorage.setItem('zipcode', zipcode)
          setIsLoading(false);
          props.history.push(`/profile/${id}`)
        })
        .catch(err => {
          setErrorMessage("Uh Oh, Looks like that didn't work. Try Again!")
          setIsLoading(false)
        });
  }

  return (
    <>
      <Dimmer active={ isLoading ? true : false }>
        <Loader>Loading</Loader>
      </Dimmer>
      <Grid textAlign='center' style={{ height: '100vh' }} verticalAlign='middle'>
        <Grid.Column style={{ maxWidth: 450 }}>
          <Image src={Welcome} centered size='medium' />
          <Header as='h2' color='teal' textAlign='center'>
            Log in to your account
          </Header>
          <Header as="h4" color='green' textAlign='center'>{props.message}</Header>
          <Header as="h4" color='red' textAlign='center'>{errorMessage}</Header>
          <Form size='large' onSubmit={loginHandler}>
            <Segment stacked>
              <Form.Input
                fluid icon='user'
                name="username"
                value={inputData.username}
                onChange={handleInput}
                iconPosition='left'
                placeholder='Username'
              />
              <Form.Input
                fluid
                icon='lock'
                iconPosition='left'
                placeholder='Password'
                type='password'
                name="password"
                value={inputData.password}
                onChange={handleInput}
              />
              <Button type="submit" color='facebook' fluid size='large'>
                Log in
              </Button>
            </Segment>
          </Form>
          <Message>
            New to us?
            <Button className="register-button"
            onClick={()=> props.history.push('/register')}
            content='Sign Up'
            positive
            size='mini' />
          </Message>
        </Grid.Column>
      </Grid>
    </>
  )
};

export default Login
