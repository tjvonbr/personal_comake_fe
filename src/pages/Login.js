import React, { useContext, useState } from 'react';
import axios from 'axios';
import { Dimmer, Loader } from 'semantic-ui-react';
import LoginGrid from '../components/LoginGrid';
import { AuthContext } from '../context/AuthContext';

function Login(props) {
  const authContext = useContext(AuthContext);

  const [credentials, setCredentials] = useState({
    username: '',
    password: '',
  });
  const [loading, setLoading] = useState(false);

  // Handle input
  function handleInput(event) {
    setCredentials({ ...credentials, [event.target.name]: event.target.value });
  }

  // Login user
  async function loginHandler(event) {
    try {
      event.preventDefault();
      setLoading(true);
      // Handles the login and storage of auth credentials
      const { data } = await axios
        .post('http://localhost:4000/auth/login', credentials)
      authContext.setAuthInfo(data);
      const { id } = data.userInfo;
      setLoading(false);
      props.history.push(`profile/${id}`);
    } catch (error) {
      console.log(error)
    }
  }

  return (
    <div className='login-main-container'>
      <Dimmer active={ loading ? true : false }>
        <Loader>Loading...</Loader>
      </Dimmer>
      <LoginGrid />
      <div className='login-form-wrapper'>
        <a 
          href="/"
          className='login-form-wrapper-header'
        >
          Vase
        </a>
        <div className='login-form-content-wrapper'>
          <div className='login-form-content'>
            <form onSubmit={loginHandler}>
              <h3 className='login-form-header'>
                Please sign in to your account
              </h3>
              <label className='login-form-label'>
                Username
                <input
                  id="usernameInput"
                  className='login-form-input'
                  type="text"
                  name="username"
                  onChange={handleInput}
                  value={credentials.username}
                />
              </label>
              <label htmlFor="password" className='login-form-label'>Password</label>
              <input
                id="passwordInput"
                className='login-form-input'
                type="password"
                name="password"
                onChange={handleInput}
                value={credentials.password} />
              <button id="btn-login">
                Login
              </button>
            </form>
            <p 
              className='login-form-register-link'>
                Still need an account? 
              <a href="/register"> Register </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Login;
