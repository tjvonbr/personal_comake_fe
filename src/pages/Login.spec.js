import React from 'react';
import axios from 'axios';
import {Login} from './Login'
import renderer from 'react-test-renderer';

test('Login renders correctly', () => {
	const login = renderer.create(<Login />);
	let JSONlogin = login.toJSON();
	expect(JSONlogin).toMatchSnapshot();
})