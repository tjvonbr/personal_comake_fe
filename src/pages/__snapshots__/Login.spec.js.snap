// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="login-main-container"
>
  <div
    className="ui dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui text loader"
      >
        Loading...
      </div>
    </div>
  </div>
  <div
    className="css-f5hm5o-LoginGrid"
  >
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span
      className="css-g17t6r-LoginGrid"
    />
    <span
      className="css-g17t6r-LoginGrid"
    />
    <span
      className="css-g17t6r-LoginGrid"
    />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span
      className="css-l6cq0a-LoginGrid"
    />
    <span
      className="css-l6cq0a-LoginGrid"
    />
    <span
      className="css-l6cq0a-LoginGrid"
    />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span
      className="css-17omzbg-LoginGrid"
    />
    <span
      className="css-17omzbg-LoginGrid"
    />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span
      className="css-444w4b-LoginGrid"
    />
    <span
      className="css-2nq4jk-LoginGrid"
    />
    <span
      className="css-2nq4jk-LoginGrid"
    />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span
      className="css-2nq4jk-LoginGrid"
    />
    <span
      className="css-2nq4jk-LoginGrid"
    />
    <span />
    <span />
  </div>
  <div
    className="login-form-wrapper"
  >
    <a
      className="login-form-wrapper-header"
      href="/"
    >
      Vase
    </a>
    <div
      className="login-form-content-wrapper"
    >
      <div
        className="login-form-content"
      >
        <form
          onSubmit={[Function]}
        >
          <h3
            className="login-form-header"
          >
            Please sign in to your account
          </h3>
          <label
            className="login-form-label"
          >
            Username
            <input
              className="login-form-input"
              id="usernameInput"
              name="username"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label
            className="login-form-label"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="login-form-input"
            id="passwordInput"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <button
            id="btn-login"
          >
            Login
          </button>
        </form>
        <p
          className="login-form-register-link"
        >
          Still need an account?
          <a
            href="/register"
          >
             Register 
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
